### PR TITLE
Move extensions from sld to core

### PR DESCRIPTION
**Pull request by colinepiloquet**
Signed-off-by: Coline PILOQUET <coline.piloquet@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
The extensions from single-line diagram repository are moved to powsybl-core.

**What is the current behavior?** *(You can also link to an open issue here)*
The extensions are in single-line diagram repository.

**What is the new behavior (if this is a feature change)?**
The extensions are in powsybl-core repository.

### CODE REVIEW

**Review by miovd — APPROVED** [review]

**Review by miovd — CHANGES_REQUESTED** [review]
Could you refactor the serializers (by using `AbstractExtensionXmlSerializer`) to lower duplication?

**Comment by geofjamg** [comment]
> Could you refactor the serializers (by using `AbstractExtensionXmlSerializer`) to lower duplication?

Also this is the right moment to replace deprecated  getExtensionWriter

File: cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CouplingDeviceDiagramData.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import com.powsybl.iidm.network.TwoWindingsTransformer;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author Massimo Ferraro <massimo.ferraro@techrain.eu>
+ */
+public class CouplingDeviceDiagramData<T extends Identifiable<T>> extends AbstractExtension<T> {
```
Comment (miovd) [inline cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CouplingDeviceDiagramData.java:21]: Should we update this to use the new extension mechanism?

File: cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/VoltageLevelDiagramData.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.VoltageLevel;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.*;
+
+/**
+ * @author Christian Biasuzzi <christian.biasuzzi@techrain.eu>
+ */
+public final class VoltageLevelDiagramData extends AbstractExtension<VoltageLevel> {
```
Comment (miovd) [inline cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/VoltageLevelDiagramData.java:18]: Same remark

**Review by miovd — COMMENTED** [review]
Some javadoc should be added

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java [hunk 3]
```diff
@@ -21,10 +21,16 @@ default String getName() {
         return NAME;
     }
 
+    /**
+     * The position of the busbar relative to other busbars.
```
Comment (annetill) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java:25]: What is a bus bar definition here ? We only have `busBarSection` objects in IIDM. Can you be more precise, maybe adding vertical or horizontal positions and the term "in general" for the interpretation if you prefer.

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java [hunk 4]
```diff
@@ -21,10 +21,16 @@ default String getName() {
...
+     */
     int getBusbarIndex();
 
     BusbarSectionPosition setBusbarIndex(int busbarIndex);
 
+    /**
+     * The position of the section relative to the other section of the busbar.
```
Comment (annetill) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java:32]: What is a section here ? We only have `busBarSection` objects in IIDM. Same remark as before.

Comment (geofjamg) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java:32]: as it is described it is a section of a busbar (busbarSection)

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java (hunk 3 above)
Comment (geofjamg) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java:25]: it is busbar section

Comment (geofjamg) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/BusbarSectionPosition.java:25]: We should not speak about vertical or horizontal, it is just an interpretation of SLD drawing.

**Review by geofjamg — CHANGES_REQUESTED** [review]
As already discussed, *DiagramData are not related to CGMES and should not be in CGMES specific extension package. It has to be refactored to be moved to general iidm extension.

File: cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CouplingDeviceDiagramData.java (hunk 1 above)
Comment (geofjamg) [reply cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CouplingDeviceDiagramData.java]: yes

**Review by annetill — APPROVED** [review]

**Pull request merged**